plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -603,6 +616,9 @@ private List<DeltaLakeColumnHandle> getColumns(MetadataEntry deltaMetadata)
     @Override
     public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTableHandle tableHandle)
     {
+        if (tableHandle instanceof CorruptedDeltaLakeTableHandle corruptedTableHandle) {
+            throw corruptedTableHandle.createException();
+        }
```

Comment:
nit: you can extract `throwIfCourrupted(ConnectorTableHandle handle)` method

Reply:
i wanted to have `throw` in the calling code so that code flow analysis works, but i don't think there is much gain from that.

will change